src/clarity-angular/data/datagrid/datagrid-pagination.ts

```diff
@@ -19,12 +19,10 @@ import {Subscription} from "rxjs/Subscription";
                 <button (click)="page.current = 1">1</button>
             </li>
             <li *ngIf="page.current > 3">...</li>
-            <li *ngIf="page.current > 1">
-                <button (click)="page.previous()">{{page.current - 1}}</button>
-            </li>
-            <li class="pagination-current">{{page.current}}</li>
-            <li *ngIf="page.current < page.last">
-                <button (click)="page.next()">{{page.current + 1}}</button>
+            <li *ngFor="let pageNum of middlePages" [ngClass]="{'pagination-current': pageNum === page.current}">
```

Comment:
When you need to do a simple class binding like this, 
```
[class.pagination-current]="pageNum === page.current"
```
is slightly more optimized, it doesn't instantiate a directive.